src/osd/ReplicatedPG.cc

```diff
@@ -3254,22 +3246,35 @@ ReplicatedPG::RepGather *ReplicatedPG::trim_object(const hobject_t &coid)
       new_snaps.insert(*i);
   }
 
+  vector<snapid_t>::iterator p = snapset.clones.end();
+
+  if (new_snaps.empty())
+  {
+    p = find(snapset.clones.begin(), snapset.clones.end(), coid.snap);
```

Comment:
please use std::find (we have using namespace std currentyl, but that will hopefully get cleaned up someday)